shark_turbine/kernel/wave/indexing.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass
+from typing import Optional, Any
+from ...support.logging import get_logger
+from .._support.indexing import IndexExpr, IndexSymbol
+import sympy
+
+logger = get_logger("turbine.wave.indexing")
+
+
+@dataclass
+class IndexSequence:
+    start: IndexExpr | int
+    size: IndexExpr | int
+    stride: Optional[IndexExpr | int] = 1
+
+    def __add__(self, other: Any) -> Any:
```

Comment:
Not really intuitive to me, when would we add index_sequence together? Can we add a small comment on that?

Reply:
The index sequence represents an access pattern and this is attempting to define arithmetic on access patterns. Currently, this is only being used when computing indices from constraints. Not sure whether it will be used for other things. But the idea is this: Say our access pattern is for workgroup i to read from offset b * i and it reads 1 tile of size b with a stride of b. Next, wave j reads from offset c * j and it reads 4 elements with stride of 1. So when we combine these access patterns, we get that wave j needs to read 4 elements from b * i + c * j, with a stride of 1. But I do agree this is non-intuitive, so let me try and see if we can remove this altogether.

Reply:
Okay got rid of it in the latest patch.